Implementation of the index design

germanruzca opened the pull request:
# Description
What: The design from mockup was implemented (header and main page), it was used scss framework to the html files, also use inputs, buttons and labels, addition of scss components, use of global variables and Font Aweosme and Google Fonts tools. 
Why: With these changes is finished the main page view, to the user/employee.

See issue #6 for more background.
 ## Screenshots
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/43076503/157779998-5df62b5c-f6db-4856-b441-3a813ec7fe3f.png">

germanruzca committed:
Change the title in the main html page.

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Use of import correctly in application.scss

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
The global variables were created

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Creation of the button css style

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Implementation of header styles

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Implementation of the styles to index page

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Global variables

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Construction of the index html file

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Add Font Awesome tool

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Structure of the header html file

Co-authored-by: Nallely Garcia  <nallely64@users.noreply.github.com>

germanruzca committed:
Change the css and html heder files

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Create the global css styles to the headings

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Change order of the files imported

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Import the main font Roboto

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Creation of the button styles

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Move the clock icon to the middle

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Implement the styles in css and structure in html

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Delete the button styles from the helper file

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Addition of color variables

Red, white, wheat, green, black;

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

JuanequeX commented on app/assets/stylesheets/base/_globals.scss:
```diff
@@ -0,0 +1,13 @@
+p {
+  font-family: 'Roboto', sans-serif;
+  font-weight: 300;
+}
```
Add a blank line here

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/components/_buttons.scss:
```diff
@@ -0,0 +1,9 @@
+  font-size: 20px;
+  height: 50px;
+  width: 200px;
+  cursor: pointer;
```
sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/components/_buttons.scss:
```diff
@@ -0,0 +1,9 @@
+  height: 50px;
+  width: 200px;
+  cursor: pointer;
+}
```
Add blank line here

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/layout/_header.scss:
```diff
@@ -0,0 +1,34 @@
+    transition: all .4s ease-in-out;
+    font-weight: 700;
+    flex: 0 1 auto;
+    margin-left: auto;      
```
sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/layout/_header.scss:
```diff
@@ -0,0 +1,34 @@
+    flex: 0 1 auto;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/layout/_header.scss:
```diff
@@ -0,0 +1,34 @@
+  position: relative;  
+  padding: 0 20px;
+  width: 100%;
+  align-items: center;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/layout/_header.scss:
```diff
@@ -0,0 +1,34 @@
+      background-color: $red_button_hover;
+    }
+  }
+}
```
Add blank line here

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+.main {
+  background-color: $blue_background;
+  height: 85vh;
+  display: flex;
+  justify-content: space-around;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+    justify-content: center;
+    margin: 0;
+    font-family: $font;
+  
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+  
+    .hour {
+      font-size: 300px;
+      color:  $black_hour_color;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+
+    .date {
+      font-size: 60px;
+      color: $black_hour_color;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+      font-size: 45px;
+      margin-bottom: 10px;
+      font-family: $font;
+      color: $aqua_label;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+      font-family: $font;
+      border: 0;
+      border-radius: 10px;
+      padding: 10px;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+      margin: 20px;
+      font-family: $font;
+      font-size: 30px;
+      align-self: center;
```
Sort this block

germanruzca replied:
Done

JuanequeX commented on app/assets/stylesheets/pages/_index.scss:
```diff
@@ -0,0 +1,75 @@
+  }
+  
+  .secondary-button-style {
+      background-color: $red_button;
+      color: $wheat_button;
+      font-weight: 700;
```
Sort this block

germanruzca replied:
This block is already sort

JuanequeX commented on app/views/home/index.html.erb:
```diff
@@ -1,2 +1,19 @@
+      CHECK
+    </button>
+  </div>
+</div>
```
Add a blank line here

germanruzca replied:
Done

JuanequeX commented on app/views/layouts/_header.html.erb:
```diff
@@ -0,0 +1,6 @@
+  <button class="primary-button-style">
+    Admin Tools
+  </button>
+</div>
```
Add a blank line here

germanruzca replied:
Done

germanruzca committed:
Required changes fixed

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

YilKanda commented on app/views/layouts/application.html.erb:
```diff
@@ -1,16 +1,19 @@
 
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
+    <link rel="stylesheet" href="path/to/font-awesome/css/font-awesome.min.css">
```
Is this line of code functional?

germanruzca replied:
I just checked, is not needed, it was removed. Thank you!

germanruzca committed:
No last blank line in application.scss

germanruzca committed:
Prueba de ultima linea Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Remove last line from test Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Remove the link tag from application.html.erb

Was removed because the app works fine just with the final script to implement the Font Awesome Icons.

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

germanruzca committed:
Sort last block from index.scss

Co-authored-by: nallely64 <nallelygar.64@gmail.com>

JuanequeX reviewed (approved):
LGTM

juperameza reviewed (approved):
LGTM :) ♥

YilKanda reviewed (approved):
LGTM!

germanruzca committed:
Merge branch 'master' into feature#6/Design-Implemented
